### ISSUE TITLE
[java] FinalFieldCouldBeStatic false positive for array initializers

### ISSUE THREAD

**Issue by zbynek** [pmd/pmd#5858]
**Affects PMD Version:** 
7.11 to 7.15 (current)

**Rule:**

Please provide the rule name and a link to the rule documentation:
[FinalFieldCouldBeStatic](https://docs.pmd-code.org/pmd-doc-7.14.0/pmd_rules_java_design.html#finalfieldcouldbestatic)

**Description:**
Suggesting that final fields of array type should be static is generally wrong, unless we also check that there are no mutations.
This is correctly ensured for `final double coordinates = new double[]{0, 0}` but not `final double coordinates = {0, 0}`.

**Code Sample demonstrating the issue:**

```java

public class Point {
final double coordinates = {0, 0};

public void setCoordinate(int index, double value) {
coordinates[index] = value;
}

```

**Expected outcome:**

PMD reports a violation at line `final double coordinates = new double[]{ 0, 0}`, but that's wrong. 

**Running PMD through:** 

Gradle

**Notes**
This behavior changed in https://github.com/pmd/pmd/commit/4cf51a6e6c2663c45cc190f4da35504ff7671adf as part of #5553.
